src/executor/copy_from_file.rs

```diff
@@ -78,14 +79,9 @@ impl CopyFromFileExecutor {
         let mut iter = reader.records();
         let mut finished = false;
         while !finished {
-            // create array builders
-            let mut array_builders = self
-                .plan
-                .logical()
-                .column_types()
-                .iter()
-                .map(|ty| ArrayBuilderImpl::with_capacity(PROCESSING_WINDOW_SIZE, ty))
-                .collect_vec();
+            // create chunk builders
+            let mut chunk_builder =
+                DataChunkBuilder::new(self.plan.logical().column_types(), PROCESSING_WINDOW_SIZE);
 
             // read records and push to array builder
             for _ in 0..PROCESSING_WINDOW_SIZE {
```

Comment:
I think `chunk_builder` can be created outside `while !finished` loop, and `for _ in 0..PROCESSING_WINDOW_SIZE` loop is not needed any more?

Reply:
+1

Reply:
Ok, I will try to fix it. I am not so familiar with rust. 
Sorry for that.

Reply:
No need to be sorry. Just try it! 😄 

You may also refer to other refactors e.g. https://github.com/risinglightdb/risinglight/pull/623/files to see how the pattern changes

Reply:
Thx for your patience! It's of great help:)